Tests/TABResourceLoaderTests/Protocols/NetworkDataResourceServiceURLSessionTests.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  NetworkDataResourceServiceURLSessionTests.swift
+//  TABResourceLoader
+//
+//  Created by Sam Dods on 12/01/2017.
+//  Copyright © 2017 Luciano Marisi. All rights reserved.
+//
+
+import XCTest
+import TABResourceLoader // note this is intentionally *not* @testable, so we can confirm URLSession is exposed
+
+class MockSessionToTestExposed: URLSessionType {
+  func perform(request: URLRequest, completion: @escaping (Data?, URLResponse?, Error?) -> Void) {
+    completion(Data(), nil, nil)
+  }
+}
+
+class NetworkDataResourceServiceURLSessionTests: XCTestCase {
+  
+  struct MockResource: NetworkResourceType, DataResourceType {
+    typealias Model = String
+    var url = URL(string: "test")!
+    func result(from data: Data) -> Result<Model> {
+      return .success("done")
+    }
+  }
+  
+  func test_initWithURLSession() {
```

Comment:
it might seem like this test is a bit over the top, and testing things that are tested elsewhere. but what i'm trying to validate is that this can be used publicly without any issues. it's almost just testing that it compiles.